Storage/SQL/SQLiteHistory.swift

```diff
@@ -451,46 +451,28 @@ extension SQLiteHistory: BrowserHistory {
             args = []
         }
 
-        let ungroupedSQL = [
-        "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, title, guid, domain_id, domain,",
-        "COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate,",
-        "COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate,",
-        "COALESCE(count(\(TableVisits).is_local), 0) AS visitCount",
-        "FROM \(TableHistory)",
-        "INNER JOIN \(TableDomains) ON \(TableDomains).id = \(TableHistory).domain_id",
-        "INNER JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id",
+        let sql = [
+        "SELECT",
+            "history.id AS historyID, history.url, title, guid, domain_id, domain,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 1 THEN visits.date ELSE 0 END), 0) AS localVisitDate,",
+            "COALESCE(MAX(CASE visits.is_local WHEN 0 THEN visits.date ELSE 0 END), 0) AS remoteVisitDate,",
+            "COALESCE(COUNT(visits.is_local), 0) AS visitCount",
+            includeIcon ? ", iconID, iconURL, iconDate, iconType, iconWidth" : "",
+        "FROM",
+            "history",
+                "INNER JOIN domains ON domains.id = history.domain_id",
+                "INNER JOIN visits ON visits.siteID = history.id",
+                includeIcon ? "LEFT OUTER JOIN view_history_id_favicon ON view_history_id_favicon.id = history.id" : "",
         whereClause,
         "GROUP BY historyID",
-        ].joined(separator: " ")
-
-        let historySQL = [
-        "SELECT historyID, url, title, guid, domain_id, domain, visitCount,",
-        "max(localVisitDate) AS localVisitDate,",
-        "max(remoteVisitDate) AS remoteVisitDate",
-        "FROM (", ungroupedSQL, ")",
-        "WHERE (visitCount > 0)",    // Eliminate dead rows from coalescing.
-        "GROUP BY historyID",
-        "ORDER BY max(localVisitDate, remoteVisitDate) DESC",
+        "HAVING COUNT(visits.is_local) > 0",
```

Comment:
If I'm reading this right, this is a huge change in behavior — you're eliminating remote-only history from the history panel. If I visit a new page on my desktop, sync, then sync my phone, the remote visit won't appear in history.

That'll definitely make the history panel faster, but also makes it pretty useless.

You already have an inner join on `visits`, so it should not be necessary to prune unvisited history.